### PR TITLE
Support function references

### PR DESCRIPTION
**Pull request by SerVB**
Found that a simple case is unsupported. +36 box tests pass.

### CODE REVIEW

File: python/box.tests/reports/git-history-plot.svg
Comment (krzema12) [inline python/box.tests/reports/git-history-plot.svg:65]: Looks like the MicroPython series didn't get an increase. It's because the plot is generated only for regular Python:

https://github.com/krzema12/kotlin-python/blob/af0c7cfbacc9fa1cc9bec5c8e68847b3946dc33a/.github/workflows/generate_reports.main.kts#L120

Merging independently generated plots for both pythons into one would mean some non-trivial effort. The good news is that with the next commit to `python-backend` branch, it should update the MicroPython series because the new failed-tests.txt for MP will be accessible from within the regular Python's reports generation task, so it's not a critical issue. Just pointing it out in case you want to retrigger it manually after merging, for the sake of keeping both series up-to-date :)

@edit: or maybe even it should fix itself after re-running the reports generation job on the PR branch? Running it now: https://github.com/krzema12/kotlin-python/actions/runs/1900265958

**Review by krzema12 — APPROVED** [review]
:tada: Thank you! :taco:

File: python/box.tests/reports/git-history-plot.svg
Comment (SerVB) [reply python/box.tests/reports/git-history-plot.svg:65]: I see the added commit is empty and nothing is updated in the plot.

Do you think it can be solved from the workflow side or should we change the generator for that?

Comment (krzema12) [reply python/box.tests/reports/git-history-plot.svg:65]: Good question, but I don't know and have no resources to investigate it today, sorry :( feel free to merge it and track it as a bug, to unblock us.

**Pull request merged**
